### ISSUE TITLE
Plans to support more custom endpoints in provider AWS (lambda, apigateway, ...)

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#2171]
_This issue was originally opened by @ebarault as hashicorp/terraform#16544. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Hi,

Following discussion started in https://github.com/hashicorp/terraform/issues/6535 (which is closed now)

From the doc, it seems that provider AWS does not yet support some endpoints customization in the nested endpoints block such as lambda and apigateway.

e.g.
```
provider "aws" {
  endpoints {
    s3 = "http://localhost:4572"
    # lambda = ...
    # apigateway = ...
  }
}
```

I was planning to test localstack against my terraform project but these endpoints are missing currently.

It would be nice to match the localstack service coverage !

Are there plan to add the ability to customize more endpoints in AWS provider?

**Comment by kprzerwa-r7** [comment]
custom endpoints for more aws services? yes please!

**Comment by bflad** [comment]
Support for custom endpoints for the below have been released in [terraform-provider-aws version 1.7.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#170-january-12-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

* acm
* apigateway
* ecr
* ecs
* lambda
* sts
* route53
